libs/langchain/langchain/chains/combine_documents/stuff.py

```diff
@@ -1,19 +1,40 @@
 """Chain that combines documents by stuffing into context."""
-
 from typing import Any, Dict, List, Optional, Tuple
 
 from langchain_core.documents import Document
 from langchain_core.prompts import BasePromptTemplate, format_document
 from langchain_core.prompts.prompt import PromptTemplate
 from langchain_core.pydantic_v1 import Extra, Field, root_validator
+from langchain_core.runnables import Runnable, RunnableParallel
 
 from langchain.callbacks.manager import Callbacks
-from langchain.chains.combine_documents.base import (
-    BaseCombineDocumentsChain,
-)
+from langchain.chains.combine_documents.base import BaseCombineDocumentsChain
 from langchain.chains.llm import LLMChain
 
 
+def create_stuff_documents_chain(
+    llm_chain: Runnable,
+    document_input_key: str,
+    *,
+    document_prompt: Optional[BasePromptTemplate] = None,
+    document_separator: str = "\n\n",
+    output_key: Optional[str] = None,
+) -> Runnable:
+    document_prompt = document_prompt or _get_default_document_prompt()
+
+    def format_documents(inputs: dict) -> dict:
+        inputs[document_input_key] = document_separator.join(
+            format_document(doc, document_prompt) for doc in inputs[document_input_key]
+        )
+        return inputs
+
+    chain = format_documents | llm_chain
+    if output_key:
```

Comment:
Do we actually want two separate output types? Feels like this just adds complexity